mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/V1ToV2MigrationThread.java

```diff
@@ -76,9 +76,7 @@ public void run() {
     }
 
     private Optional<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>> poll() {
-        synchronized (messagesToBeMigrated) {
-            return Optional.ofNullable(messagesToBeMigrated.poll());
-        }
+        return Optional.ofNullable(messagesToBeMigrated.poll());
```

Comment:
Replace poll by take please (poll will ntot block for ever but return after a certain ammount of time) take will block forever until there is data!